sponsorship_compassion/model/contracts.py

```diff
@@ -581,83 +540,76 @@ def contract_waiting(self, cr, uid, ids, context=None):
                 for line in contract.contract_line_ids:
                     sponsorship = line.sponsorship_id
                     if sponsorship.state == 'active':
-                        cr.execute(
+                        self.env.cr.execute(

```

Comment:
From new API guidelines, I read we should call `self.env.invalidate_all()` after doing queries with the cursor in order to update then environment.